src/main/java/com/google/firebase/projectmanagement/FirebaseProjectManagementServiceImpl.java

```diff
@@ -0,0 +1,909 @@
+
+  @Override
+  public ShaCertificate createShaCertificate(String appId, String shaHash)
+      throws FirebaseProjectManagementException {
+    return createShaCertificateOp(appId, shaHash).call();
+  }
+
+  @Override
+  public ApiFuture<ShaCertificate> createShaCertificateAsync(String appId, String shaHash) {
+    return createShaCertificateOp(appId, shaHash).callAsync(app);
+  }
+
+  private CallableOperation<ShaCertificate, FirebaseProjectManagementException>
+      createShaCertificateOp(final String appId, final String shaHash) {
+    return new CallableOperation<ShaCertificate, FirebaseProjectManagementException>() {
+      @Override
+      protected ShaCertificate execute() throws FirebaseProjectManagementException {
+        String url = String.format(
+            "%s/v1beta1/projects/-/androidApps/%s/sha", FIREBASE_PROJECT_MANAGEMENT_URL, appId);
+        ShaCertificateResponse parsedResponse = new ShaCertificateResponse();
+        ImmutableMap<String, String> payload = ImmutableMap.<String, String>builder()
+            .put("sha_hash", shaHash)
+            .put("cert_type", ShaCertificate.getTypeFromHash(shaHash).name())
+            .build();
+        makePostRequest(url, payload, parsedResponse, appId, "App ID");
+        return ShaCertificate.create(
+            parsedResponse.name, parsedResponse.shaHash, parsedResponse.certType);
+      }
+    };
+  }
+
+  /* deleteShaCertificate */
+
+  @Override
+  public void deleteShaCertificate(String resourceName)
+      throws FirebaseProjectManagementException {
+    deleteShaCertificateOp(resourceName).call();
+  }
+
+  @Override
+  public ApiFuture<Void> deleteShaCertificateAsync(String resourceName) {
+    return deleteShaCertificateOp(resourceName).callAsync(app);
+  }
+
+  private CallableOperation<Void, FirebaseProjectManagementException> deleteShaCertificateOp(
+      final String resourceName) {
+    return new CallableOperation<Void, FirebaseProjectManagementException>() {
+      @Override
+      protected Void execute() throws FirebaseProjectManagementException {
+        String url = String.format(
+            "%s/v1beta1/%s", FIREBASE_PROJECT_MANAGEMENT_URL, resourceName);
+        EmptyResponse parsedResponse = new EmptyResponse();
+        makeDeleteRequest(url, parsedResponse, resourceName, "SHA name");
+        return null;
+      }
+    };
+  }
+
+  private static class ListShaCertificateResponse {
+    @Key("certificates")
+    private List<ShaCertificateResponse> certificates;
+  }
+
+  // This class is public due to the way parsing nested JSON objects work, and is needed by
+  // getShaCertificates. In any case, the containing class, FirebaseProjectManagementServiceImpl, is
+  // package-private.
+  public static class ShaCertificateResponse {
+    @Key("name")
+    private String name;
+
+    @Key("shaHash")
+    private String shaHash;
+
+    @Key("certType")
+    private String certType;
+  }
+
+  /* Helper methods. */
+
+  private <T> void makeGetRequest(
```

Comment:
Consider introducing a new helper class. May be `HttpHelper.java`?

Reply:
No, not a project-wide helper class. Just a helper for this package/class. Basically get all the http cruft out of this class.